File: lib/java/projecteuler/Series.java
package projecteuler;

import java.lang.Math;

public class Series {

  public static int sumNaturals(int n) {
    return (n * (n + 1)) / 2;
  }

  public static double sumGeometric(double a, double n, double r) {
    return a * (1 - Math.pow(r, n)) / (1 - r);
  }
}

File: lib/scala/projecteuler/Series.scala
package projecteuler

object Series {
  def sumNaturals(n: Int): Int = {
    return (n * (n + 1)) / 2
  }

  def sumGeometric(a: Double, n: Double, r: Double): Double = {
    return a * (1 - Math.pow(r, n)) / (1 - r);
  }
}
